src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -294,6 +293,61 @@ public bool IsSolutionAvailable
             }
         }
 
+        public IEnumerable<string> GetDeferredProjectsFilePath()
+        {
+#if VS14
+            // Not applicable for Dev14 so always return empty list.
+            return Enumerable.Empty<string>();
+#else
+            return ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                List<string> projectPaths = new List<string>();
+                IEnumHierarchies enumHierarchies;
+                Guid guid = Guid.Empty;
+                _vsSolution.GetProjectEnum((uint)__VSENUMPROJFLAGS3.EPF_DEFERRED, ref guid, out enumHierarchies);
+
+                // Loop all projects found
+                if (enumHierarchies != null)
+                {
+                    // Loop projects found
+                    IVsHierarchy[] hierarchy = new IVsHierarchy[1];
+                    uint fetched = 0;
+                    while (enumHierarchies.Next(1, hierarchy, out fetched) == VSConstants.S_OK && fetched == 1)
+                    {
+                        string projectPath;
+                        hierarchy[0].GetCanonicalName(VSConstants.VSITEMID_ROOT, out projectPath);
+
+                        if (!string.IsNullOrEmpty(projectPath))
+                        {
+                            projectPaths.Add(projectPath);
+                        }
+                    }
+                }
+
+                return projectPaths;
+            });
+#endif
+        }
+
+        public bool SolutionHasDeferredProjects()
+        {
+#if VS14
+            // for Dev14 always return false since DPL not exists there.
+            return false;
+#else
+            return ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    
+                object value;
+                _vsSolution.GetProperty((int)(__VSPROPID7.VSPROPID_DeferredProjectCount), out value);
```

Comment:
Please add a comment on this. Or even better add an utility method (we have many occurrences of similar code).